### PR TITLE
[RLLib] Fix not choosing next obs from original space, but from preprocessed space.

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
Signed-off-by: Artur Niederfahrenhorst <artur@anyscale.com>

## Why are these changes needed?

In EnvRunnverV2, if all agents are done, we sample a dummy all-zeros next obs from the observation space of the policy, rather than the original space that the ObsPreprocessingConnector expects.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/evaluation/env_runner_v2.py [hunk 1]
```diff
@@ -637,8 +638,12 @@ def _process_observations(
                     policy_id: PolicyID = episode.policy_for(agent_id)
                     policy = self._worker.policy_map[policy_id]
 
-                    # Create a fake (all-0s) observation.
-                    obs_space = policy.observation_space
+                    # Create a fake (all-0s) observation (from the original space if
+                    # available
+                    obs_space = (
+                        policy.agent_connectors[ObsPreprocessorConnector]
```
Comment (kouroshHakha) [inline rllib/evaluation/env_runner_v2.py:644]: what are you indexing agent_connectors with? the class? This line does not make sense to me.

Comment (ArturNiederfahrenhorst) [reply rllib/evaluation/env_runner_v2.py:644]: Yes, you can do this. It makes sense.

Comment (kouroshHakha) [reply rllib/evaluation/env_runner_v2.py:644]: what is the content of `policy.agent_connectors` ?

Comment (ArturNiederfahrenhorst) [reply rllib/evaluation/env_runner_v2.py:644]: I modified the __get__ method of the pipeline a while back! have a look 😄

Comment (ArturNiederfahrenhorst) [reply rllib/evaluation/env_runner_v2.py:644]: ObsPreprocessorConnector, Viewrequirements and State Buffer?

Comment (kouroshHakha) [reply rllib/evaluation/env_runner_v2.py:644]: oh shit :) what is the expected return value of `policy.agent_connectors[ObsPreprocessorConnector]`?

Comment (ArturNiederfahrenhorst) [reply rllib/evaluation/env_runner_v2.py:644]: A list! Just fixed fixed this line thanks for asking!

Comment (ArturNiederfahrenhorst) [reply rllib/evaluation/env_runner_v2.py:644]: So we take the first observationspreprocessor and get the original space from there.

Comment (ArturNiederfahrenhorst) [reply rllib/evaluation/env_runner_v2.py:644]: -> This PR is a little "drafty". I'm yet to test this and maybe will write a unit test.

**Comment by ArturNiederfahrenhorst** [comment]
I just let CI run to test if this breaks "everything" in some weird way. Disabling connectors on this PR again now. @gjoliver

File: rllib/evaluation/env_runner_v2.py (hunk 1 above)
Comment (gjoliver) [reply rllib/evaluation/env_runner_v2.py:644]: add a little bit of error handling? what if ObsPreprocessorConnector is not used?

**Comment by kouroshHakha** [comment]
closing this in favor of this #30602

**Pull request closed without merging**